sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py

```diff
@@ -48,27 +49,32 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-def _parse_conn_str(conn_str):
-    # type: (str) -> Tuple[str, Optional[str], Optional[str], str, Optional[str], Optional[int]]
+def _parse_conn_str(conn_str, check_case=False):
+    # type: (str, Optional[bool]) -> Tuple[str, Optional[str], Optional[str], str, Optional[str], Optional[int]]
     endpoint = None
     shared_access_key_name = None
     shared_access_key = None
     entity_path = None  # type: Optional[str]
     shared_access_signature = None  # type: Optional[str]
     shared_access_signature_expiry = None  # type: Optional[int]
-    for element in conn_str.strip().split(";"):
-        key, _, value = element.partition("=")
-        if key.lower() == "endpoint":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "hostname":
-            endpoint = value.rstrip("/")
-        elif key.lower() == "sharedaccesskeyname":
-            shared_access_key_name = value
-        elif key.lower() == "sharedaccesskey":
-            shared_access_key = value
-        elif key.lower() == "entitypath":
-            entity_path = value
-        elif key.lower() == "sharedaccesssignature":
+
+    # split connection string into properties
+    conn_settings = [s.split("=", 1) for s in conn_str.rstrip(";").split(";")]
```

Comment:
This looks neat! :P
may I ask why we want to rstrip trailing ";" here, as we previously just have `conn_str.strip()`.

Reply:
I saw that in [this test case](https://github.com/Azure/azure-sdk-for-python/blob/db4892a29c45fd9d65da1319d65ffd3269f13898/sdk/servicebus/azure-servicebus/tests/test_sb_client.py#L259), the manually built conn str ends with `;`. I would have changed the test case if we had the assumption that customers can only get the conn str from the portal (where conn strings don't end in `;`). However, since users can manually create the conn strings, they may end the connection string with `;`. This would add an empty string to `conn_settings` when splitting on `;` and raise the `ValueError`. So I kept the trailing `;` in the test case, and accounted for it here. Would it have been better to change the test case instead, and say that a connection string cannot end with `;`?

Reply:
I think we should change the test case instead, keeping `conn_str.strip()` first.
we didn't know whether a conn_str with trailing ';' is valid or not, it's better the keep the status quo until we get more input :P

Reply:
It looks like .NET does strip trailing `;` : https://github.com/Azure/azure-sdk-for-net/blob/5fedc94bca9fb9d1a941eda7a8af1eceebb5809a/sdk/servicebus/Microsoft.Azure.ServiceBus/tests/ServiceBusConnectionStringBuilderTests.cs#L75. For consistency, should we also `rstrip(';')`. So something like `conn_str.strip().rstrip(';').split(';')`.

Reply:
I think we're good now, thanks for addressing my bikeshedding!